Update echo to version 3

joelnb opened the pull request:
I believe this pull request does everything required to enable the use of the latest version (v3) of github.com/labstack/echo which has an API different to the current version in use. It is inspired by the pull request from @rtuin and built on top of the work by @phuongvu to update to v2.

Currently I have added a version constraint on `^3.0.0` for echo to the `glide.yaml` (to show that it is not compatible with earlier versions and to hopefully enable easier use of `glide update` in future) but I can remove that if you would prefer to just rely on the `glide.lock` for controlling the version of dependencies.

This pull request also update the version of all other dependencies as this was required to use the new version of echo (hence the massive number of changes).

olebedev commented:
@joelnb, it's better to rely on the glide.lock.

joelnb commented:
Okay, I pushed another commit which removes that from the `glide.yaml` & updates the `glide.lock`.

olebedev commented on server/react.go:
```diff
@@ -65,8 +63,8 @@ func (r *React) Handle(c echo.Context) error {
 	case re := <-vm.Handle(map[string]interface{}{
-		"url":     c.Request().(*standard.Request).URI(),
-		"headers": c.Request().(*standard.Request).Request.Header,
+		"url":     c.Request().URL.Path,
```
This should be a `URL.String()` instead of `URL.Path`.

joelnb commented:
@olebedev thanks for the review. I corrected that error and also added 2 more commits which remove the gzip middleware (as you suggested in https://github.com/olebedev/go-starter-kit/pull/55) and revert the `e` variable to being called `engine` as it seems a more descriptive name (and is the name currently in master anyway so reduces the changes made here).

olebedev commented:
Thanks!

rtuin commented:
Very nice, thanks!

olebedev commented:
@rtuin could you leave some thoughts here #68 ?
